pkg/deployment/deployment.go

```diff
@@ -168,38 +168,33 @@ func (d *Deployment) send(ev *deploymentEvent) {
 func (d *Deployment) run() {
 	log := d.deps.Log
 
-	if d.status.State == api.DeploymentStateNone {
+	if d.status.Phase == api.DeploymentPhaseNone {
 		// Create secrets
 		if err := d.resources.EnsureSecrets(); err != nil {
-			d.failOnError(err, "Failed to create secrets")
-			return
+			d.CreateEvent(k8sutil.NewErrorEvent("Failed to create secrets", err, d.GetAPIObject()))
 		}
 
 		// Create services
 		if err := d.resources.EnsureServices(); err != nil {
-			d.failOnError(err, "Failed to create services")
-			return
+			d.CreateEvent(k8sutil.NewErrorEvent("Failed to create services", err, d.GetAPIObject()))
```

Comment:
Does `d.CreateEvent` somehow leave this function early (maybe with a panic)? If not, is it right to simply move on, for example if the services have not been setup, should we go on to pods?

Reply:
CreateEvent just comes back.
The inspection loop will keep trying to create all resources, so if something fails here, we just keep trying later.